docs-sphinx/environment.yml.cog

```diff
@@ -3,8 +3,8 @@ channels:
   - conda-forge
 dependencies:
   - pip
-  - xeus-cling=0.13.0
-  - gxx_linux-64=9.4.0
+  - cxx-compiler
+  - root
   - pip:
     # [[[cog
     # import cog
```

Comment:
It's best to remove, rather than comment-out, unnecessary code. It's always available in history.

Also, if this doesn't need to be an environment.yml.cog anymore, but can be a static environment.yml, that would also be preferred. I was nervous about adding it in the first place because it's like replacing pyproject.toml with a setup.py: configurations are safer and more maintainable when they're declarative/static, and most of the work in the Python configuration world is trying to get more declarative/static. This is a backwards step.

(When you need it, you need it, so it's nice that such things exist, but we should view it as having a negative weight. Now that it's not needed—great! Let's go back to static files!)

Reply:
(I see: it's not commented _out_; commenting is just how cog works.)

Also, I see that this environment.yml (for docs) is derived from the global requirements.txt, and that's why you're using cog here.

I'm not objecting to cog as a particular technology to make that happen; but rather that we're doing it at all. Sometimes, two copies of some code/configuration that are clearly supposed to be the same thing is better than a clever system to remove one duplication. Actually, I start thinking about duplication when the number gets up to three or four.

What we really want here is some declarative syntax for "insert some other file here", like setup.cfg has, replacing the `open("source-file", "r").read()` that used to be in setup.py. Because that _is_ what we're talking about here with cog: going from a setup.cfg-like solution to a setup.py-like solution.

Aha! It looks like there is such a syntax: https://stackoverflow.com/a/59056234/1623645

Reply:
Okay, sure. Let's stick with cog, at least until Mamba is fixed.